app/src/main/java/com/belatrixsf/connect/ui/contacts/ContactsListActivity.java

```diff
@@ -44,4 +47,13 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         setNavigationToolbar();
     }
 
+    public static Intent makeIntent(Context context) {
+        return new Intent(context, ContactsListActivity.class);
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        Constants.PARENT_ACTIVITY_INDEX = Constants.PARENT_USER;
```

Comment:
You are creating an static variable in constants which you are modifying it on a back pressed? Can you describe the problem? at least a not so bad approach would be check the instance of the activity to ContactList but still that's a bad approach.